components/style/matching.rs

```diff
@@ -635,6 +627,40 @@ pub trait MatchMethods : TElement {
         }
     }
 
+    /// Updates the style attribute rule nodes without re-running selector
+    /// matching, using just the rule tree.
+    fn update_style_attribute(&self,
```

Comment:
Given that we're going to be stacking this up with similar calls (for animations and transitions, etc), it seems like this should just accept the old rule node as an argument and return a rule node, rather than operating on ElementData and returning MatchResults. We can handle grabbing the old pseudo rule nodes at the callsite, once all the rule tree fixups have been performed.